src/main/groovy/com/netflix/gradle/plugins/application/OspackageApplicationSpringBootPlugin.groovy

```diff
@@ -54,85 +53,95 @@ class OspackageApplicationSpringBootPlugin implements Plugin<Project> {
     void apply(Project project) {
         project.plugins.apply(OspackageApplicationPlugin)
 
-        if (!project.plugins.hasPlugin('org.springframework.boot')) {
-            throw new IllegalStateException("The 'org.springframework.boot' plugin must be applied before applying this plugin")
+        project.afterEvaluate {
+            if (!project.plugins.hasPlugin('org.springframework.boot')) {
+                project.logger.error("The '{}' plugin requires the '{}' plugin.",
+                        "com.netflix.nebula.ospackage-application-spring-boot",
+                        "org.springframework.boot")
+                throw new RuntimeException("The 'com.netflix.nebula.ospackage-application-spring-boot' plugin requires the 'org.springframework.boot' plugin.")
+            }
         }
 
-        // Spring Boot 2.0 configured distributions that have everything we need
-        if (project.distributions.findByName('boot') != null) {
-            // Use the main distribution and configure it to have the same baseName as the boot distribution
-            project.tasks.named(JavaPlugin.JAR_TASK_NAME) {
-                enabled = true
-            }
-            project.afterEvaluate {
-                project.tasks.named('bootJar').configure {
-                    if (GradleVersion.current().baseVersion < GradleVersion.version('6.0').baseVersion) {
-                        classifier = 'boot'
-                    } else {
-                        archiveClassifier = 'boot'
-                    }
+        project.plugins.withId("org.springframework.boot") {
```

Comment:
withId allows the plugins to be listed in any order, and this block will always execute after the spring boot plugin is applied

Reply:
most of the rest of this diff is just from the extra indentation... it looks like much more than it actually is